### DISCUSSION TITLE
x-slot in child component

### DISCUSSION THREAD

**Discussion by yassermohammad** [livewire/livewire#5017] [1. Help]
Hello and thanks for good framework,
I have two childs component inside my parent component, here:
index.blade.php
![Screenshot from 2022-06-07 21-29-57](https://user-images.githubusercontent.com/34732441/172440633-1523dadb-9f17-4854-bdd9-12663bb5a250.png)

I added x-slot script tag which contains an event click listener on any button in the step-one like this
step-one.blade.php
![Screenshot from 2022-06-07 21-40-18](https://user-images.githubusercontent.com/34732441/172442396-14847bb7-d92e-4ef9-9c96-a09f477fb495.png)

I added a this script's name in the default layout view(I called it "master.blade.php") like this:
{{ $script ?? '' }}

but it does not attached to it,
how can I listen to click on the button in a child component by jquery script?

**Comment by joshhanley** [comment]
@yassermohammad I would use `@stack` in your layout and `@push` in your component for pushing the scripts to the layout file instead of trying to use slots, as slots only work up one layer.

Hope this helps!
